drivers/kw2xrf: Use TMR3 instead TMR4 to detect ACK timeout

Ilmu011 opened the pull request:
_Originally posted by @PeterKietzmann in #12083_
> @Ilmu011 for the TMR3 vs TMR4 discussion, please keep that one separate in an own issue or even better in a fixing PR :-)

This driver currently uses TMR4 to detect ACK timeouts in TR operations. There is a dedicated timer for detecting timeouts in RX-operations and that is TMR3. I am working on #10364 right now and the kw41z driver uses TMR3 for this purpose, which seems to make more sense.

See [section 7.6.2](https://www.nxp.com/docs/en/reference-manual/MKW2xDxxxRM.pdf#page=171) of the Reference Manual "Using T3CMP to Abort an RX operation"

benpicco commented on drivers/kw2xrf/kw2xrf_netdev.c:
```diff
@@ -694,21 +695,29 @@ static void _isr_event_seq_tr(netdev_t *netdev, uint8_t *dregs)
+            DEBUG("[kw2xrf] TC3TMOUT, SEQIRQ, TX failed\n");
+            irqsts1 |= MKW2XDM_IRQSTS1_SEQIRQ;
+            assert(dev->pending_tx != 0);
+            dev->pending_tx--;
```
The above three lines are common in both cases, you can put them before the `if`.

Ilmu011 replied:
I'll fix this.

benpicco commented on drivers/kw2xrf/kw2xrf_netdev.c:
```diff
@@ -694,21 +695,29 @@ static void _isr_event_seq_tr(netdev_t *netdev, uint8_t *dregs)
+            netdev->event_callback(netdev, NETDEV_EVENT_TX_NOACK);
+            kw2xrf_timer3_seq_abort_off(dev);
+            kw2xrf_abort_rx_ops_disable(dev);
+            kw2xrf_set_sequence(dev, dev->idle_state);
```
I guess this is not the same as `kw2xrf_set_idle_sequence()`?

Ilmu011 replied:
This is how it originally was before. According to the comments, `kw2xrf_set_idle_sequence()` is specifically for handling interrupts? There is a lot of code duplication with `kw2xrf_set_sequence` too.

Ilmu011 replied:
Also, line 705-707 could be de-duplicated too.

Ilmu011 commented:
@benpicco It looks like the difference between `kw2xrf_set_idle_sequence()` and  `kw2xrf_set_sequence` is that `kw2xrf_set_idle_sequence()` checks whether there are still pending transmissions and doesn't abort the sequence if that is the case.
I'm using `kw2xrf_set_idle_sequence()` now.

@PeterKietzmann I also removed the dupilcate code and added some comments.

benpicco commented on drivers/kw2xrf/kw2xrf_netdev.c:
```diff
@@ -694,21 +697,24 @@ static void _isr_event_seq_tr(netdev_t *netdev, uint8_t *dregs)
+            /* Disable interrupt for TMR3 and reset TMR3IRQ */
+            kw2xrf_abort_rx_ops_disable(dev);
+            /* Go back to idle state */
+            kw2xrf_set_idle_sequence(dev);
```
You forgot to bring the indentation back one level 😉

benpicco commented on drivers/kw2xrf/kw2xrf_netdev.c:
```diff
@@ -694,21 +697,24 @@ static void _isr_event_seq_tr(netdev_t *netdev, uint8_t *dregs)
+            /* Disable interrupt for TMR3 and reset TMR3IRQ */
+            kw2xrf_abort_rx_ops_disable(dev);
+            /* Go back to idle state */
+            kw2xrf_set_idle_sequence(dev);
```
You forgot to bring the indentation back one level 😉

Ilmu011 replied:
Damnit! xD

benpicco replied:
Just squash directly.

benpicco commented:
Looks much better & easier to understand now with the smaller if blocks!

benpicco commented:
Feel free to squash.

Ilmu011 commented:
Squashed it :)

PeterKietzmann commented:
@Ilmu011 I think one commit should be enough here. Please squash all together and name it something alike the first commit.

Ilmu011 commented:
@PeterKietzmann Done.

PeterKietzmann commented:
Congrats to your first RIOT contribution!
